types/src/validator_set.rs

```diff
@@ -71,33 +66,55 @@ impl ValidatorSetResource {
         self.last_reconfiguration_time
     }
 
-    pub fn validator_set(&self) -> &ValidatorSet {
-        &self.validator_set
+    pub fn validator_set(&self) -> ValidatorSet {
+        ValidatorSet::new(self.scheme, self.validators.clone())
     }
 }
 
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[repr(u8)]
+pub enum ConsensusScheme {
+    Ed25519 = 0,
+}
+
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
-pub struct ValidatorSet(Vec<ValidatorInfo>);
+pub struct ValidatorSet {
+    scheme: ConsensusScheme,
+    payload: Vec<ValidatorInfo>,
```

Comment:
for example, in the ideal case, this could either be an enum or like this, but with the current LCS it cannot.